src/utils.ts

```diff
@@ -88,3 +93,78 @@ function parseEnvVariables(p: string) {
         return process.env[variable] || ''
     })
 }
+
+type FileLike = string | Buffer;
+export class ClipboardManager {
```

Comment:
Can we just use the [`ClipboardItem`](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem/ClipboardItem) API instead? Browser support isn't great, but for the majority of users that isn't relevant.

Generally I'm very uncomfortable with shipping binaries here.

Reply:
> Generally I'm very uncomfortable with shipping binaries here.

So do I, but I tried several other ways and nothing worked.

The browser support isn't the problem, Clipboard Item [doesn't support SVGs ](https://stackoverflow.com/questions/62390200/domexception-sanitized-mime-type-image-svgxml-not-supported-on-write). Also, Electron has an experimental clipboard manager but the Electron API isn't accessible from the extension.

Reply:
Mhh. IMHO either converting to png or just using the source string are better than shipping binaries, even though  both of those options kinda suck too.

Reply:
To get the clipboard to work reliably with remote environments the [VSCode clipboard API](https://code.visualstudio.com/api/advanced-topics/remote-extensions#using-the-clipboard) must be used which only supports text via `readText`/`writeText` which makes it impossible to copy an SVG or a PNG.

Reply:
Are you sure about that? JS running in an in-editor iframe should have access to the clipboard, given that these iframes are created with `allow="clipboard-read; clipboard-write;"`. You just need to use the VSCode clipboard API when you want to copy something out of the extension host process.

Reply:
>Historically, extension authors have used Node.js modules such as clipboardy to interact with the clipboard. Unfortunately, if you use these modules in a Workspace Extension, they will use the remote clipboard instead of the user's local one. The VS Code clipboard API solves this problem. It is always run locally, regardless of the type of extension that calls it.

I might understand this wrong.
So should I go with png then?

Reply:
Yeah, the important difference here is where the extension code runs.
"Normal" extension code (e.g. all the TS code in this extension) will run on the remote machine (in an extension host process), so the clipboard that's available from there is not the one the user has immediate access to. JS in the iframe on the other hand always runs on your local machine/browser, so it should have access to your clipboard as well.

So imho it's worth investigating that solution, yes.

Reply:
Now, it uses the Clipboard API.